From 9ae1e1d49f14a257f7ab24d7578ab121ef87443f Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Thu, 2 Nov 2017 15:01:54 +0800
Subject: [PATCH 2744/5242] MLK-16693-2: mipi_csi: fix null video name setting

commit  3efbf7d385a4609435788a888bb41ea9f7b1aeb9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Setting video device name before memset operation
lead to null value of vdev->name

Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit d342872a4db74cdbbc955b0e5c73d3cd86b217be)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-cap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-cap.c b/drivers/media/platform/imx8/mxc-isi-cap.c
index ae8813e..10d2334 100644
--- a/drivers/media/platform/imx8/mxc-isi-cap.c
+++ b/drivers/media/platform/imx8/mxc-isi-cap.c
@@ -1340,8 +1340,8 @@ static int mxc_isi_register_cap_device(struct mxc_isi_dev *mxc_isi,
 	int ret = -ENOMEM;
 
 	dev_dbg(&mxc_isi->pdev->dev, "%s\n", __func__);
-	snprintf(vdev->name, sizeof(vdev->name), "mxc_isi.%d.capture", mxc_isi->id);
 	memset(vdev, 0, sizeof(*vdev));
+	snprintf(vdev->name, sizeof(vdev->name), "mxc_isi.%d.capture", mxc_isi->id);
 
 	vdev->fops	= &mxc_isi_capture_fops;
 	vdev->ioctl_ops	= &mxc_isi_capture_ioctl_ops;
-- 
1.7.9.5

